### ISSUE TITLE
Add support for Pyright as a quality-checking tool

### ISSUE THREAD

**Issue by dperl-dls** [Bachmann1234/diff_cover#379]
I'm interested in using this tool to check our diffs so that we can migrate some projects to using Pyright type checking (stricter than the mypy we are currently using) without having to fix all of our errors at once or having our CI broken for a long time. I hope others would also find this useful. I'm happy to add the `driver` for it as long as I could count on someone being available to review it some time.

**Comment by Bachmann1234** [comment]
Yep, im still here and do reviews. This is a solo project that I maintain on personal time so I cant promise im always avaliable. But Ill communicate if I think its gonna be a bit before I can review

**Comment by dperl-dls** [comment]
Yeah, that's totally understandable, thanks!
